### ISSUE TITLE
IColumn currentWidth doesn't reflect the rendered width of the column

### ISSUE THREAD

**Issue by isabelacmor** [microsoft/fluentui#11515]
### Environment Information

- **Package version(s)**: ^7.64.1

### Please provide a reproduction of the bug in a codepen:

[https://codepen.io/isabelacmor/pen/VwYPgwa?editors=0010](https://codepen.io/isabelacmor/pen/VwYPgwa?editors=0010)

#### Actual behavior:

`IColumn`'s `currentWidth` doesn't reflect the width of the rendered column. As you can see in the CodePen, the TextFields rendered are the wrong width, so they don't line up with the columns. For example, if you look at the "Date Modified" column the TextField rendered is 90px, but the column itself is 134px.

#### Expected behavior:

`IColumn`'s `currentWidth` should reflect the item's actual width. Or, provide a way to override the column headers to allow for additional UI to be rendered alongside the existing UI. If I handle rendering the column headers manually, I have to provide all the functionality for reordering columns, clicking, icons to show sort order, etc. All I want to do is add more UI alongside what already exists.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Normal

**Comment by ecraig12345** [comment]
From the documentation, it looks like `currentWidth` was intended for internal use only, so it really shouldn't even be in the public interface (also the DetailsList shouldn't be mutating its props...but that's a larger design issue). It was probably added for some specific scenario and may not work predictably in other scenarios. So I'm hesitant to invest in debugging and improving its behavior--especially since that would probably be harder than it sounds on the surface.

However, I do think it would be reasonable to add a prop in either `IDetailsHeaderProps` or maybe `IColumn` to customize rendering of DetailsHeader cell content, similar to other `onRender____` props. This was previously requested in #1814 but the discussion didn't go anywhere.

**Comment by ecraig12345** [comment]
I also wonder if there's some way to achieve what you need with CSS. @micahgodbolt might be more help in providing suggestions there.

**Comment by ecraig12345** [comment]
Actually never mind, a CSS solution (which doesn't rely on `columnWidth`) probably isn't possible since the TextField isn't inside any of the cells, and the "columns" are just a bunch of elements set to the same width. I was thinking of it as if the control actually used table-style layout, which it doesn't.

**Comment by isabelacmor** [comment]
> However, I do think it would be reasonable to add a prop in either `IDetailsHeaderProps` or maybe `IColumn` to customize rendering of DetailsHeader cell content, similar to other `onRender____` props. This was previously requested in #1814 but the discussion didn't go anywhere.

I saw that issue and think it'd be a great (and reasonable) solution to this. Consumers of DetailsList can already customize other content with respective `onRender____` props, so I was surprised to learn this wasn't the case for DetailsHeader.

I was a bit confused to see `currentWidth` being exposed since the comments say it's internal. :) But I also tried getting this to work with `onColumnResize` and noticed that the `newWidth` that function provides is the same value as `currentWidth`(which isn't the actual width of the column).
